avm/res/document-db/database-account/main.bicep

```diff
@@ -0,0 +1,504 @@
+    eventHubName: diagnosticSetting.?eventHubName
+    metrics: diagnosticSetting.?metricCategories ?? [
+      {
+        category: 'AllMetrics'
+        timeGrain: null
+        enabled: true
+      }
+    ]
+    logs: diagnosticSetting.?logCategoriesAndGroups ?? [
+      {
+        categoryGroup: 'AllLogs'
+        enabled: true
+      }
+    ]
+    marketplacePartnerId: diagnosticSetting.?marketplacePartnerResourceId
+    logAnalyticsDestinationType: diagnosticSetting.?logAnalyticsDestinationType
+  }
+  scope: databaseAccount
+}]
+
+resource databaseAccount_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for (roleAssignment, index) in (roleAssignments ?? []): {
+  name: guid(databaseAccount.id, roleAssignment.principalId, roleAssignment.roleDefinitionIdOrName)
+  properties: {
+    roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : contains(roleAssignment.roleDefinitionIdOrName, '/providers/Microsoft.Authorization/roleDefinitions/') ? roleAssignment.roleDefinitionIdOrName : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName)
+    principalId: roleAssignment.principalId
+    description: roleAssignment.?description
+    principalType: roleAssignment.?principalType
+    condition: roleAssignment.?condition
+    conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+    delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+  }
+  scope: databaseAccount
+}]
+
+module databaseAccount_sqlDatabases 'sql-database/main.bicep' = [for sqlDatabase in sqlDatabases: {
+  name: '${uniqueString(deployment().name, location)}-sqldb-${sqlDatabase.name}'
+  params: {
+    databaseAccountName: databaseAccount.name
+    name: sqlDatabase.name
+    containers: contains(sqlDatabase, 'containers') ? sqlDatabase.containers : []
+    throughput: contains(sqlDatabase, 'throughput') ? sqlDatabase.throughput : 400
+    autoscaleSettingsMaxThroughput: contains(sqlDatabase, 'autoscaleSettingsMaxThroughput') ? sqlDatabase.autoscaleSettingsMaxThroughput : -1
+  }
+}]
+
+module databaseAccount_mongodbDatabases 'mongodb-database/main.bicep' = [for mongodbDatabase in mongodbDatabases: {
+  name: '${uniqueString(deployment().name, location)}-mongodb-${mongodbDatabase.name}'
+  params: {
+    databaseAccountName: databaseAccount.name
+    name: mongodbDatabase.name
+    collections: contains(mongodbDatabase, 'collections') ? mongodbDatabase.collections : []
+  }
+}]
+
+module databaseAccount_gremlinDatabases 'gremlin-database/main.bicep' = [for gremlinDatabase in gremlinDatabases: {
+  name: '${uniqueString(deployment().name, location)}-gremlin-${gremlinDatabase.name}'
+  params: {
+    databaseAccountName: databaseAccount.name
```

Comment:
Could it be that for this and other child resources the majority of parameters isn't passed down?

Reply:
good catch, you are right. Fixed in ead490de661e1e5653f8bbe906c28d2a7c0935af